tiledb/sm/query/writer.h

```diff
@@ -75,10 +75,9 @@ class Writer : public StrategyBase, public IQueryStrategy {
      * offsets tile, whereas the second tile is the values tile. In both cases,
      * the third tile stores a validity tile for nullable attributes.
      */
-    std::unordered_map<
-        std::string,
-        std::tuple<WriterTile, WriterTile, WriterTile>>
-        last_tiles_;
+    std::unordered_map<std::string, std::vector<WriterTile>> last_tiles_;
```

Comment:
The problem is more extensive here. The whole Writer class uses vector of tiles where tile[0] is the fixed sized buffer (either offsets or data), tile[1] will be either the var data or nullable data, and tile[2] might be nullable data. This all has to change IMO for a better solution. For now, last_tiles_ is a vector of 'WriterTile' because it allows it to be used inside of the existing functions. Changing from a tuple to a vector removed the need to make a shallow clone so while not optimal, it is still an improvement in this to be soon improved version.

Reply:
https://app.shortcut.com/tiledb-inc/story/14539/writer-find-better-solution-when-dealing-with-vector-of-tiles